src/datasources/network/fetch.network.service.ts

```diff
@@ -18,37 +18,37 @@ export class FetchNetworkService implements INetworkService {
     private readonly loggingService: ILoggingService,
   ) {}
 
-  async get<T>(
-    baseUrl: string,
-    { params, ...options }: NetworkRequest = {},
-  ): Promise<NetworkResponse<T>> {
-    const url = this.buildUrl(baseUrl, params);
+  async get<T>(args: {
+    url: string;
+    networkRequest?: NetworkRequest;
+  }): Promise<NetworkResponse<T>> {
+    const url = this.buildUrl(args.url, args.networkRequest?.params);
     const startTimeMs = performance.now();
     try {
       return await this.client<T>(url, {
         method: 'GET',
-        ...options,
+        headers: args.networkRequest?.headers,
       });
     } catch (error) {
       this.logErrorResponse(error, performance.now() - startTimeMs);
       throw error;
     }
   }
 
-  async post<T>(
-    baseUrl: string,
-    data: object,
-    { params, headers }: NetworkRequest = {},
-  ): Promise<NetworkResponse<T>> {
-    const url = this.buildUrl(baseUrl, params);
+  async post<T>(args: {
+    url: string;
+    data: object;
+    networkRequest?: NetworkRequest;
+  }): Promise<NetworkResponse<T>> {
+    const url = this.buildUrl(args.url, args.networkRequest?.params);
     const startTimeMs = performance.now();
     try {
       return await this.client<T>(url, {
         method: 'POST',
-        body: JSON.stringify(data),
+        body: JSON.stringify(args.data),
         headers: {
           'Content-Type': 'application/json',
```

Comment:
Unrelated with this PR but I think that the content type should be driven entirely by the `NetworkRequest`.

Reply:
Agree. I'll make a note to address this in a follow up at some stage.